FRFuel.cs

```diff
@@ -17,68 +16,39 @@ public class FRFuel : BaseScript
-        public static string manualRefuelAnimDict = "weapon@w_sp_jerrycan";
+        private static string fuelLevelPropertyName = "_Fuel_Level";
 
-        public static string[] tankBones = new string[] {
+        private static List<Pickup> _jerryCans = new List<Pickup>();
+        private static string manualRefuelAnimDict = "weapon@w_sp_jerrycan";
+        private static InLoopOutAnimation _jerryCanAnimation = new InLoopOutAnimation(
+               new Animation(manualRefuelAnimDict, "fire_intro"),
+               new Animation(manualRefuelAnimDict, "fire"),
+               new Animation(manualRefuelAnimDict, "fire_outro")
+             );
+
+        private static string[] tankBones = new string[] {
             "petrolcap",
             "petroltank",
             "petroltank_r",
             "petroltank_l",
             "wheel_lr"
         };
 
-        protected Blip[] blips;
-        protected List<Pickup> pickups;
-
-        protected float fuelTankCapacity = 65f;
-
-        protected float refuelRate = 1f;
-        protected float fuelConsumptionRate = 1f;
-        protected float fuelAccelerationImpact = 0.0002f;
-        protected float fuelTractionImpact = 0.0001f;
-        protected float fuelRPMImpact = 0.0005f;
-
-        public float showMarkerInRangeSquared = 250f;
-
-        public HUD hud;
-        public Random random = new Random();
-
-        private int currentGasStationIndex;
-        private Vehicle lastVehicle;
+        private static Random _random = new Random();
+        private static bool hudActive = false;
 
-        protected bool currentVehicleFuelLevelInitialized = false;
-        protected bool hudActive = false;
-        protected bool refuelAllowed = true;
-        protected float addedFuelCapacitor = 0f;
+        private static float addedFuelCapacitor = 0f;
+        private bool refuelAllowed = true;
 
-        protected InLoopOutAnimation jerryCanAnimation;
-
-        protected Vehicle LastVehicle { get => lastVehicle; set => lastVehicle = value; }
-
-        protected Config Config { get; set; }
-        protected bool showHud = true;
-        protected bool showHudWhenEngineOff = true;
-
-        protected bool initialized = false;
         public static Control engineToggleControl = Control.VehicleHorn;
-
-        private string fuelBarNormalHexColor = "FFB300"; // 255, 179, 0
-        private string fuelBarWarningHexColor = "FFF5DC"; // 255, 179, 0
         #endregion
 
         /// <summary>
         /// Ctor
         /// </summary>
         public FRFuel()
         {
-            hud = new HUD();
-
-            jerryCanAnimation = new InLoopOutAnimation(
-              new Animation(manualRefuelAnimDict, "fire_intro"),
-              new Animation(manualRefuelAnimDict, "fire"),
-              new Animation(manualRefuelAnimDict, "fire_outro")
-            );
+            EventHandlers["onClientResourceStart"] += new Action<string>(OnClientResourceStart);
```

Comment:
This event can't be relied on as it has some instabilities, we were relying on it first too, but decided to not to.

Reply:
Good feedback